python/ray/ray_operator/operator.py

```diff
@@ -18,18 +19,17 @@
 from ray.ray_operator import operator_utils
 from ray.ray_operator.operator_utils import AUTOSCALER_RETRIES_FIELD
 from ray.ray_operator.operator_utils import STATUS_AUTOSCALING_EXCEPTION
-from ray.ray_operator.operator_utils import STATUS_ERROR
 from ray.ray_operator.operator_utils import STATUS_RUNNING
 from ray.ray_operator.operator_utils import STATUS_UPDATING
 from ray import ray_constants
 
 logger = logging.getLogger(__name__)
 
 # Queue to process cluster status updates.
-cluster_status_q = mp.Queue()  # type: mp.Queue[Tuple[str, str, str]]
```

Comment:
This queue is used in an `mp.Process` here 
https://github.com/ray-project/ray/blob/cbc24f43a5d4cf0bb533b19a4bf6ecd2ff3aba16/python/ray/ray_operator/operator.py#L80 

I think `queue.Queue` is thread-safe but it might not work well with subprocesses.
Should we stick with the `mp.Queue`?

Reply:
In the current design I don't expect to see subprocesses being created. This would break the existing state such as  the collection of `RayCluster` objects. There's more overhead to using a multiprocessing queue, so I wouldn't use it unless it's needed. Is there a specific usage you had in mind here?

Reply:
Sorry, this is a bit tangled:

Currently RayCluster's `create_or_update` method runs bound method `self._create_or_update`  in a subprocess.
`self._create_or_updates` launches the head and runs the autoscaling monitor in the `try` block -- the `except` block is accessed when the monitor fails and puts "autoscaler retrying" status into the queue.

So the queue is accessed from a subprocess if the monitor fails..  

I realize now this setup is pretty bad, as the `RayCluster` object is probably serialized in some terrible way to be passed into the subprocess.

Reply:
Ah, didn't notice that usage. Okay, changed it back to multiprocessing queue. Will be nice once the autoscaling stuff is moved into its own Pod.

Reply:
> Will be nice once the autoscaling stuff is moved into its own Pod.
100%